The `_find_product_fixed_metadata` method does not always return same data

Unclear if this is somehow caused by the same thing that triggers #742, but here's a failed test run in the CI for a PR that only did CI-changes for other workflows:
```
__________________ test_sampled_product_fixed_fields[default] __________________

summary_store = <cubedash.summary._stores.SummaryStore object at 0x7efc0c2d2d50>

    def test_sampled_product_fixed_fields(summary_store: SummaryStore) -> None:
        # Compute fixed fields using a sampled percentage.
    
        # (We're doing this manually to force it to use table sampling -- our test data is
        # not big enough to trigger it in the `run_generate()` tests)
    
        # Tiled product, sampled
        fixed_fields = summary_store._find_product_fixed_metadata(
            summary_store.index.products.get_by_name("ga_ls9c_ard_3"),
            sample_datasets_size=5,
        )
    
>       assert fixed_fields == {
            "platform": "landsat-9",
            "instrument": "OLI_TIRS",
            "product_family": "ard",
            "format": "GeoTIFF",
            "eo_gsd": 15.0,
            "label": None,
            "dataset_maturity": "final",
        }
E       assert Documents differ:
E             {'dictionary_item_removed': ["root['fmask_snow']"]}
E         Full output document: 
E         {'format': 'GeoTIFF', 'label': None, 'eo_gsd': 15.0, 'platform': 'landsat-9', 'fmask_snow': 0.0, 'instrument': 'OLI_TIRS', 'product_family': 'ard', 'dataset_maturity': 'final'}

integration_tests/test_summarise_data.py:340: AssertionError
```
I think the document diff is done backwards, so the method returned something with `fmask__snow` and that isn't expected by the test.

--- a/integration_tests/test_summarise_data.py
+++ b/integration_tests/test_summarise_data.py
@@ -345,7 +345,7 @@ def test_sampled_product_fixed_fields(summary_store: SummaryStore) -> None:
     product = summary_store.index.products.get_by_name("ga_ls9c_ard_3")
     assert product is not None
     fixed_fields = summary_store._find_product_fixed_metadata(
-        product, sample_datasets_size=5
+        product, sample_datasets_size=6
     )
 
     assert fixed_fields == {

